sdk/trace/src/test/java/io/opentelemetry/sdk/trace/testbed/clientserver/TestClientServerTest.java

```diff
@@ -56,12 +56,25 @@ void test() throws Exception {
 
     List<SpanData> finished = otelTesting.getSpans();
     assertThat(finished).hasSize(2);
-
+    int flag = 1;
     finished = TestUtils.sortByStartTime(finished);
-    assertThat(finished.get(0).getTraceId()).isEqualTo(finished.get(1).getTraceId());
-    assertThat(finished.get(0).getKind()).isEqualTo(SpanKind.CLIENT);
-    assertThat(finished.get(1).getKind()).isEqualTo(SpanKind.SERVER);
 
+    if (finished.get(0).getKind().equals(SpanKind.CLIENT)) {
+      flag = 2;
+    }
+
+    if (flag == 2) {
+      assertThat(finished.get(0).getTraceId()).isEqualTo(finished.get(1).getTraceId());
+      assertThat(finished.get(0).getKind()).isEqualTo(SpanKind.CLIENT);
+      assertThat(finished.get(1).getKind()).isEqualTo(SpanKind.SERVER);
+    } else {
+      SpanData r = finished.get(0);
+      finished.set(0, finished.get(1));
+      finished.set(1, r);
+      assertThat(finished.get(0).getTraceId()).isEqualTo(finished.get(1).getTraceId());
+      assertThat(finished.get(0).getKind()).isEqualTo(SpanKind.CLIENT);
+      assertThat(finished.get(1).getKind()).isEqualTo(SpanKind.SERVER);
```

Comment:
these three lines can be pulled out of the if/else, since they are identical. 
Then, you can just skip the usage of the `flag` altogether and put the logic to swap the two spans in a simple if block.